### PR TITLE
#12716 Update version_ts for all page versions. Refresh all page vers…

### PR DESCRIPTION
**Pull request by dsilvam**
…ions from cache.

### CODE REVIEW

File: dotCMS/src/main/java/com/dotmarketing/factories/MultiTreeFactory.java [hunk 1]
```diff
@@ -668,61 +654,40 @@ public static void saveMultiTree(MultiTree o, long languageId) throws DotSecurit
...
-	    Identifier ident = APILocator.getIdentifierAPI().find(id);
-        ContentletVersionInfo versionInfo = APILocator.getVersionableAPI()
-                .getContentletVersionInfo(ident.getId(), languageId);
-        versionInfo.setVersionTs(new Date());
-        APILocator.getVersionableAPI().saveContentletVersionInfo(
-                versionInfo);
+	private static void updateHTMLPageVersionTS(String id) throws DotDataException, DotSecurityException {
+		Identifier identifier = APILocator.getIdentifierAPI().find(id);
+		List<Contentlet> allContentletVersions = APILocator.getContentletAPI()
+			.findAllVersions(identifier, APILocator.getUserAPI().getSystemUser(), false);
```
Comment (wezell) [inline dotCMS/src/main/java/com/dotmarketing/factories/MultiTreeFactory.java]: This isn't the right method - this pulls all old versions of a piece of content which can be in the 100's .  We need to pull a list of ContentletVersionInfos for a given piece of content

File: dotCMS/src/main/java/com/dotmarketing/business/VersionableFactoryImpl.java [hunk 2]
```diff
@@ -295,7 +295,17 @@ protected ContentletVersionInfo findContentletVersionInfoInDB(String identifier,
          contv = (ContentletVersionInfo)dh.load();
          return contv;
     }
+    @Override
+    protected List<ContentletVersionInfo> findAllContentletVersionInfos(final String identifier)throws DotDataException, DotStateException {
+
+         HibernateUtil dh = new HibernateUtil(ContentletVersionInfo.class);
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotmarketing/business/VersionableFactoryImpl.java]: not sure but shouldn't include cache here? if not apply nvm my comment

File: dotCMS/src/main/java/com/dotmarketing/factories/MultiTreeFactory.java [hunk 3]
```diff
@@ -668,61 +658,40 @@ public static void saveMultiTree(MultiTree o, long languageId) throws DotSecurit
...
-     * @throws DotSecurityException 
-     *            
-     */
-    private static void refreshPageInCache(String id, Long languageId) throws DotDataException, DotSecurityException {
-        Identifier ident = APILocator.getIdentifierAPI().find(id);
-        Contentlet content = APILocator.getContentletAPI()
-                .findContentletByIdentifier(ident.getId(), false, languageId, APILocator.systemUser(), false);
-        IHTMLPage htmlPage = APILocator.getHTMLPageAssetAPI().fromContentlet(content);
-        PageServices.invalidateAll(htmlPage);
+        List<Contentlet> cons = APILocator.getContentletAPIImpl().findContentlets(Lists.newArrayList(inodes));
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotmarketing/factories/MultiTreeFactory.java]: rename it to contenlets or contentList

Comment (dsilvam) [reply dotCMS/src/main/java/com/dotmarketing/factories/MultiTreeFactory.java]: Done

**Review by jdotcms — APPROVED** [review]

**Pull request merged**